### PR TITLE
feat(upload): added the upload file page

### PR DESCRIPTION
**Pull request by Shruti3004**
## Description

Added the upload file page.

## Changes

- Created the separate folder for uploads in Components that contains helper functions which will get reused in VCS, Url and server.
     - AccessLevel
     - Ignore Scm
     - License Decider
     - Optional Analysis
     - Upload Reuse
- Added the Upload file page

## How to test

visit to the route '/upload/file'

## Screenshots

Screenshot from 2021-06-28 16-01-46
Screenshot from 2021-06-28 16-24-01
Screenshot from 2021-06-29 16-51-41
Screenshot from 2021-06-29 16-51-58
Screenshot from 2021-06-29 16-52-05

### CODE REVIEW

**Comment by Shruti3004** [comment]
@GMishx current implementation is perfectly creating a file, But there is one question regarding the dropdown of select folder, I have implemented it like the way I did with search as suggested by you, by default I have provided it with the Id, is it the right approach? And for the time being we decided to wrap with all the `uploads` api then we will work on jobs api for scanning the uploaded documents, What's your call on this?

**Comment by GMishx** [comment]
> @GMishx current implementation is perfectly creating a file, But there is one question regarding the dropdown of select folder, I have implemented it like the way I did with search as suggested by you, by default I have provided it with the Id, is it the right approach?

You can check #45, @Aman-Codes is loading them.

> And for the time being we decided to wrap with all the `uploads` api then we will work on jobs api for scanning the uploaded documents, What's your call on this?

Okay, lets check the REST API and discuss this in the meeting. There is a possibility to do it in single call.

**Comment by Shruti3004** [comment]
> > @GMishx current implementation is perfectly creating a file, But there is one question regarding the dropdown of select folder, I have implemented it like the way I did with search as suggested by you, by default I have provided it with the Id, is it the right approach?
> 
> You can check #45, @Aman-Codes is loading them.
> 
> > And for the time being we decided to wrap with all the `uploads` api then we will work on jobs api for scanning the uploaded documents, What's your call on this?
> 
> Okay, lets check the REST API and discuss this in the meeting. There is a possibility to do it in single call.

Okay Sure.

**Comment by Shruti3004** [comment]
@GMishx I have added the jobs api as well for scheduling analysis of uploaded files please review it.

File: src/components/Upload/CommonFields/UploadReuse/index.js
Comment (GMishx) [inline src/components/Upload/CommonFields/UploadReuse/index.js:19]: This can be fetched with combination of `/folders` and `/uploads` endpoints.

File: src/components/Upload/CommonFields/UploadReuse/index.js
Comment (GMishx) [inline src/components/Upload/CommonFields/UploadReuse/index.js:55]: The PR is merged in community. Can be included now.

File: src/components/Widgets/Input/index.js
Comment (GMishx) [inline src/components/Widgets/Input/index.js:5]: Unnecessary change. Please revert.

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by GMishx** [comment]
The file got uploaded successfully but the `uploadid` for the jobs was undefined. Also, please clear the form once the form is submitted to prevent duplicate uploads.

image

File: src/components/Widgets/Input/index.js
Comment (Shruti3004) [reply src/components/Widgets/Input/index.js]: Reverted!

File: src/api/upload.js
Comment (Shruti3004) [reply src/api/upload.js:48]: Removed the header.

File: src/Routes.js
Comment (Shruti3004) [reply src/Routes.js]: Removed the duplicate route.

**Comment by Shruti3004** [comment]
@GMishx In yesterday meeting we had a discussion regarding creating the label named urgent, actually this pr has massive changes so when I rebase it yesterday, on rebasing it overwrites the code twice, then again i copy the file from my commit. So, unamously we reached to a conclusion that we can have this label for such prs.

File: src/api/upload.js
Comment (GMishx) [reply src/api/upload.js:48]: Are you sure? It's still here.

**Comment by GMishx** [comment]
> @GMishx In yesterday meeting we had a discussion regarding creating the label named urgent, actually this pr has massive changes so when I rebase it yesterday, on rebasing it overwrites the code twice, then again i copy the file from my commit, this leads to increase the work and time wastage. So, unamously we reached to a conclusion that we can have this label for such prs.

Thank you for letting me know about the label but I can only do so much working as a single maintainer reviewing the PRs :-).

Also, there are many changes that needs to be done before I can merge this PR, so I left the review comments. You want to merge it now and resolve them in a separate issue?

File: src/constants/endpoints.js
Comment (GMishx) [inline src/constants/endpoints.js:41]: Shouldn't it be part of `jobs` object?

Comment (Shruti3004) [reply src/constants/endpoints.js:41]: Yes, we have to fix it once we hop on jobs section then we will keep all the related apis to it in that.

**Comment by Shruti3004** [comment]
> > @GMishx In yesterday meeting we had a discussion regarding creating the label named urgent, actually this pr has massive changes so when I rebase it yesterday, on rebasing it overwrites the code twice, then again i copy the file from my commit, this leads to increase the work and time wastage. So, unamously we reached to a conclusion that we can have this label for such prs.
> 
> Thank you for letting me know about the label but I can only do so much working as a single maintainer reviewing the PRs :-).
> 
> Also, there are many changes that needs to be done before I can merge this PR, so I left the review comments. You want to merge it now and resolve them in a separate issue?

Yes, I understand, @GMishx I have no probblem resolving in this pr itself but the thing is on rebasing the coding is getting overwritten, is there is any other way to resolve tht

**Comment by GMishx** [comment]
> Yes, I understand, @GMishx I have no probblem resolving in this pr itself but the thing is on rebasing the coding is getting overwritten, is there is any other way to resolve tht

Only way to prevent this situation is I do not merge any other branch till the branch marked with `urgent` tags are merged.

**Review by GMishx — APPROVED** [review]
Changes looks good.
Tested, working fine.

Just a request if the form can be reloaded or cleaned once the "Upload" is done, I can merge this PR.

**Comment by Shruti3004** [comment]
> Changes looks good.
> Tested, working fine.
> 
> Just a request if the form can be reloaded or cleaned once the "Upload" is done, I can merge this PR.

Sir I have set that wont it working for you?

**Comment by GMishx** [comment]
Oh, my bad. Didn't pull the code.

**Pull request merged**
